packages/apps/reputation-oracle/server/src/modules/user/user.service.ts

```diff
@@ -113,6 +125,53 @@ export class UserService {
     return userEntity.save();
   }
 
+  public async registerLabeler(
+    user: UserEntity,
+    data: RegisterLabelerRequestDto,
+  ) {
+    if (user.type !== UserType.WORKER) {
+      throw new BadRequestException(ErrorUser.InvalidType);
+    }
+
+    if (user.kyc?.status !== KycStatus.APPROVED) {
+      throw new BadRequestException(ErrorUser.KycNotApproved);
+    }
+
+    if (user.siteKey) {
+      return user.siteKey;
+    }
+
+    // Register user as a labeler at hcaptcha foundation
+    const registeredLabeler = await this.hcaptchaService.registerLabeler({
+      email: user.email,
+      language: 'en', // TODO: Retrieve from kyc
+      country: 'US', // TODO: Retrieve from kyc
+      address: user.evmAddress,
+    });
+
+    if (!registeredLabeler) {
+      throw new BadRequestException(ErrorUser.LabelingEnableFailed);
+    }
+
+    // Retrieve labeler site key from hcaptcha foundation
+    const labelerData = await this.hcaptchaService.getLabelerData({
+      email: user.email,
+    });
+    if (!labelerData || !labelerData.sitekeys.length) {
+      throw new BadRequestException(ErrorUser.LabelingEnableFailed);
+    }
+    const siteKey = labelerData.sitekeys[0].sitekey;
+
+    const newSiteKey = new SiteKeyEntity();
+    newSiteKey.siteKey = siteKey;
+    newSiteKey.user = user;
+    newSiteKey.type = data.oracleType;
```

Comment:
You can use `oracleType.HCAPTCHA` here instead of passing a data object, taking into consideration that you are using `hcaptchaService` here.